tools/serverpod_cli/lib/src/analyzer/dart/definition_analyzers/class_analyzer.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:path/path.dart' as path;
+import 'package:serverpod_cli/analyzer.dart';
+import 'package:serverpod_cli/src/analyzer/dart/definitions.dart';
+
+abstract class ClassAnalyzer {
+  static EndpointDefinition? analyze(
+    ClassElement element,
+    List<MethodDefinition> methodDefinitions,
+    CodeAnalysisCollector collector,
```

Comment:
Question: Is the idea to give errors here in the future? Right now the collector is not used.

Reply:
Yes, exactly. I wanted to set the pattern for the future. The following PRs will actually introduce some analysis and error messages.